pkg/vault/contracts/CompositeLiquidityRouter.sol

```diff
@@ -577,6 +611,13 @@ contract CompositeLiquidityRouter is ICompositeLiquidityRouter, BatchRouterCommo
             return 0;
         }
 
+        if (isStaticCall == false) {
+            // Take the underlying token amount, required to wrap, in advance.
+            _takeTokenIn(sender, IERC20(underlyingToken), underlyingAmountIn, wethIsEth);
+            // Since the wrap operation was paid in advance, set the underlying as settled.
+            _settledTokenAmounts().tSet(underlyingToken, underlyingAmountIn);
```

Comment:
Yes, I need to use settledTokenAmounts when the operation is EXACT_OUT, because I take all the tokens of the user in advance, wrap what we need and then I need to give the unused tokens back to the user.